docs/core/whats-new/dotnet-10/libraries.md

```diff
@@ -0,0 +1,144 @@
+
+Second, the extraction of <xref:System.IO.Compression.ZipArchive> entries is now parallelized, and internal data structures are optimized for better memory usage. These improvements address issues related to performance bottlenecks and high memory usage, making `ZipArchive` more efficient and faster, especially when dealing with large archives. Details are provided [dotnet/runtime #103153](https://github.com/dotnet/runtime/pull/103153#issue-2339713028).
+
+## Additional `TryAdd` and `TryGetValue` overloads for `OrderedDictionary<TKey, TValue>`
+
+<xref:System.Collections.Generic.OrderedDictionary`2> provides `TryAdd` and `TryGetValue` for addition and retrieval like any other `IDictionary<TKey, TValue>` implementation. However, there are scenarios where you might want to perform additional operations, so new overloads have been added that return an index to the entry:
+
+```csharp
+public class OrderedDictionary<TKey, TValue>
+{
+    // New overloads
+    public bool TryAdd(TKey key, TValue value, out int index);
+    public bool TryGetValue(TKey key, out TValue value, out int index);
+}
+```
+
+This index can then be used with <xref:System.Collections.Generic.OrderedDictionary`2.GetAt*>/<xref:System.Collections.Generic.OrderedDictionary`2.SetAt*> for fast access to the entry. An example usage of the new `TryAdd` overload is to add or update a key/value pair in the ordered dictionary:
+
+:::code language="csharp" source="../snippets/dotnet-10/csharp/snippets.cs" id="snippet2":::
+
+This new API is already used in <xref:System.Json.JsonObject> and improves the performance of updating properties by 10-20%.
+
+## Allow specifying ReferenceHandler in `JsonSourceGenerationOptions`
+
+When using source generators for JSON serialization, the generated context will throw when cycles are serialized or deserialized. This behavior can now be customized by specifying the <xref:System.Text.Json.Serialization.ReferenceHandler> in the <xref:System.Text.Json.Serialization.JsonSourceGenerationOptionsAttribute>. Here is an example using `JsonKnownReferenceHandler.Preserve`:
+
+:::code language="csharp" source="../snippets/dotnet-10/csharp/snippets.cs" id="snippet3":::
+
+## More left-handed matrix transformation methods
+
+.NET 10 adds the remaining APIs for creating left-handed transformation matrices for billboard and constrained-billboard matrices. You can use these methods like their existing right-handed counterparts [add xrefs to the existing counterparts] when using a left-handed coordinate system instead.
+
+```csharp
+public partial struct Matrix4x4
+{
+   public static Matrix4x4 CreateBillboardLeftHanded(Vector3 objectPosition, Vector3 cameraPosition, Vector3 cameraUpVector, Vector3 cameraForwardVector)
+## More left-handed matrix transformation methods
```

Comment:
Is this a TODO? Or should it be `//` rather than `##`?

Reply:
Neither, it's a merge mistake 😄